[CI]Reproducible Failure in MetaDataIndexStateServiceTests.testValidateShardLimit

```
 ./gradlew ':server:test' --tests "org.elasticsearch.cluster.metadata.MetaDataIndexStateServiceTests.testValidateShardLimit" -Dtests.seed=F9F825519FC70BF
```

fails with 

```
Suite: Test class org.elasticsearch.cluster.metadata.MetaDataIndexStateServiceTests
  1> [2019-10-01T10:58:56,101][INFO ][o.e.c.m.MetaDataIndexStateServiceTests] [testValidateShardLimit] before test
  1> [2019-10-01T10:58:56,228][INFO ][o.e.c.m.MetaDataIndexStateServiceTests] [testValidateShardLimit] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.cluster.metadata.MetaDataIndexStateServiceTests.testValidateShardLimit" -Dtests.seed=F9F825519FC70BF -Dtests.security.manager=true -Dtests.locale=kkj-CM -Dtests.timezone=Etc/GMT+0 -Dcompiler.java=12 -Druntime.java=12
  2> java.lang.AssertionError: this method will not work reliably with this many data nodes due to the limit of shards in a single index,use fewer data nodes or multiple indices
    Expected: a value less than or equal to <90>
         but: <96> was greater than <90>
        at __randomizedtesting.SeedInfo.seed([F9F825519FC70BF:341E5111B1DF7D8]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.cluster.shards.ClusterShardLimitIT$ShardCounts.forDataNodeCount(ClusterShardLimitIT.java:392)
        at org.elasticsearch.cluster.metadata.MetaDataIndexStateServiceTests.testValidateShardLimit(MetaDataIndexStateServiceTests.java:337)
  2> NOTE: leaving temporary files on disk at: /Users/armin/src/elasticsearch/server/build/testrun/test/temp/org.elasticsearch.cluster.metadata.MetaDataIndexStateServiceTests_F9F825519FC70BF-001
  2> NOTE: test params are: codec=Asserting(Lucene80): {}, docValues:{}, maxPointsInLeafNode=1383, maxMBSortInHeap=6.59391730820022, sim=Asserting(org.apache.lucene.search.similarities.AssertingSimilarity@4069e77), locale=kkj-CM, timezone=Etc/GMT+0
  2> NOTE: Mac OS X 10.14.6 x86_64/Oracle Corporation 12.0.1 (64-bit)/cpus=12,threads=1,free=384265816,total=536870912
  2> NOTE: All tests run in this JVM: [MetaDataIndexStateServiceTests]

```

every time for me locally as well as on CI.

E.g. https://gradle-enterprise.elastic.co/s/4ljb7zmb7lics/

--- a/server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateServiceTests.java
@@ -333,7 +333,7 @@ public class MetaDataIndexStateServiceTests extends ESTestCase {
     }
 
     public void testValidateShardLimit() {
-        int nodesInCluster = randomIntBetween(2,100);
+        int nodesInCluster = randomIntBetween(2, 90);
         ClusterShardLimitIT.ShardCounts counts = forDataNodeCount(nodesInCluster);
         Settings clusterSettings = Settings.builder()
             .put(MetaData.SETTING_CLUSTER_MAX_SHARDS_PER_NODE.getKey(), counts.getShardsPerNode())

